java/vector/src/main/codegen/templates/UnionListWriter.java

```diff
@@ -85,7 +84,7 @@ public int getValueCapacity() {
 
   @Override
   public void close() throws Exception {
-
+    vector.close();
```

Comment:
Should this close the `PromotableWriter`?

Reply:
promotableWriter.close only call IntWriter.close(for List<Int>) and won't clear ListVector.
vector.close() will clear all.

Reply:
I meant close the `PromotableWriter` in addition to closing the `ListVector`. It might not be necessary, but if it doesn't hurt then it's better to be safe and close both I think.

Reply:
ok, fixed.